design/README.md

```diff
@@ -0,0 +1,12 @@
+# smithy-swift Design Documents
```

Comment:
trivial: may be call this file index.md?

Reply:
I named it README.md so that it is displayed in Github under the folder contents when the folder is clicked.

https://github.com/awslabs/smithy-swift/tree/jbe/design_reorg/design